src/unix/system.c

```diff
@@ -341,7 +345,27 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (!xdg_data_home_dir) {
+	    xdg_data_home_dir = malloc(sizeof(char) * 
+				       (strlen(homedir) + strlen("/.local/share") + 1));
+	    malloc_needed = 1;
```

Comment:
You're not actually checking if the `malloc` succeeded, and can potentially double-free. Set this in a ternary

Reply:
Also, it should be 
```suggestion
	    mem_was_allocated = 1;
```
or 
```suggestion
	    need_to_free = 1;
```

Reply:
7ae9ce53